johnzon-jsonb/src/main/java/org/apache/johnzon/jsonb/JsonbAccessMode.java

```diff
@@ -1033,6 +1037,20 @@ private boolean isNumberType(final Type type) {
         return Number.class.isAssignableFrom(clazz) || clazz.isPrimitive();
     }
 
+    private boolean hasRawType(final Type type) {
+        return Class.class.isInstance(type) || ParameterizedType.class.isInstance(type);
```

Comment:
Hmm, i get the point where java.lang.reflect will highly likely return a Class (oracle jdk guarantees it) but this comes from a reader - which is configurable in the jsonb builder - so potentially a custom PT impl, not java.reflect backed.
Any issue harnessing the check (Class.isInstance(pt.getRawType()) to avoid a ClassCastException which would be a regression for end users?

Reply:
No problem with that :-)

Reply:
Will not find the time to do the changes this weekend. So go ahead, if you like. Otherwise I'll do it next week.